src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java

```diff
@@ -201,17 +213,16 @@ private Object traverseField(String fieldName, Value fieldValue) {
                 .collect(Collectors.toList());
     }
 
-    private void traverseField(IMessage message, String fieldName, Value value, IFieldStructure fieldStructure) {
-        if (fieldStructure != null) {
-            Object convertedValue = fieldStructure.isComplex()
-                    ? processComplex(value, fieldStructure)
-                    : convertSimple(value, fieldStructure);
-            message.addField(fieldName, convertedValue);
-        }
+    private void traverseField(IMessage message, String fieldName, Value value, @NotNull IFieldStructure fieldStructure) {
+        Object convertedValue = fieldStructure.isComplex()
+                ? processComplex(value, fieldStructure)
+                : convertSimple(value, fieldStructure);
+        message.addField(fieldName, convertedValue);
     }
 
     private Object convertSimple(Value value, IFieldStructure fieldStructure) {
         if (fieldStructure.isCollection()) {
+            checkKind(value, fieldStructure.getName(), KindCase.LIST_VALUE);
             return value.getListValue().getValuesList()
                     .stream()
                     .map(element -> convertToTarget(element, fieldStructure))
```

Comment:
Probably, we need to do the same as we do for the list of messages. We need to add the information about index to the error